File: backend/src/main/java/de/interaapps/pastefy/controller/stats/StatsController.java
package de.interaapps.pastefy.controller.stats;

import de.interaapps.pastefy.Pastefy;
import de.interaapps.pastefy.controller.HttpController;
import de.interaapps.pastefy.exceptions.PermissionsDeniedException;
import de.interaapps.pastefy.model.database.AuthKey;
import de.interaapps.pastefy.model.database.User;
import de.interaapps.pastefy.model.responses.app.StatsResponse;
import org.javawebstack.http.router.router.annotation.PathPrefix;
import org.javawebstack.http.router.router.annotation.params.Attrib;
import org.javawebstack.http.router.router.annotation.verbs.Get;

@PathPrefix("/api/v2/app/stats")
public class StatsController extends HttpController {
    @Get
    public StatsResponse stats(@Attrib("authkey") AuthKey authKey, @Attrib("user") User user) {
        if (!Pastefy.getInstance().getConfig().get("pastefy.publicstats", "false").equalsIgnoreCase("true") && !authKey.hasPermission("stats:read") && user.type != User.Type.ADMIN)
            throw new PermissionsDeniedException();

        return StatsResponse.create();
    }
}

File: backend-kt/src/main/kotlin/de/interaapps/pastefy/controller/stats/StatsController.kt
package de.interaapps.pastefy.controller.stats

import de.interaapps.pastefy.auth.annotations.CurrentAuthKey
import de.interaapps.pastefy.auth.annotations.CurrentUser
import de.interaapps.pastefy.dto.app.StatsResponse
import de.interaapps.pastefy.entities.AuthKey
import de.interaapps.pastefy.entities.User
import de.interaapps.pastefy.config.PastefyProperties
import de.interaapps.pastefy.exceptions.PermissionsDeniedException
import de.interaapps.pastefy.service.StatsService
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/api/v2/app/stats")
class StatsController(
    private val properties: PastefyProperties,
    private val statsService: StatsService,
) {
    @GetMapping
    fun stats(@CurrentAuthKey authKey: AuthKey?, @CurrentUser user: User?): StatsResponse {
        if (!properties.publicStats && authKey?.hasPermission("stats:read") != true && user?.isAdmin != true) {
            throw PermissionsDeniedException()
        }
        return statsService.get()
    }
}
